strax/corrections/cmt_DB_interface.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+import logging
+import os
+from bson.son import SON
+import pandas as pd
+import pdmongo as pdm
+import pymongo
+import numpy as np
+
+import strax
+
+export, __all__ = strax.exporter()
+
+@export
+class Corrections_DB_interface:
+
+    def __init__(self,host=None,username=None,password=None):
+
+        self.host = host
+        self.username = username
+        self.password = password
+        self.client = pymongo.MongoClient(host=self.host,username=self.username,password=self.password)
+        self.DATABASE_NAME= 'corrections' 
+        self.DATABASE=self.client[self.DATABASE_NAME]
+
+    def list_corrections(self):
+        """Smart logic to list corrections
+        """
+        return [x['name'] for x in self.DATABASE.list_collections() if not 'global' in x['name']]
+
+
+    def read(self, correction):
+        """Smart logic to read corrections
+        """ 
+        df = pdm.read_mongo(correction, [], self.DATABASE)
+    
+        # No data found
+        if df.size == 0:
+            return None
+        # Delete internal Mongo identifier
+        del df['_id']
+
+        # Set UTC
+        #df['time'] = pd.to_datetime(df['time'], utc=True)
+    
+        return  df.set_index('time')
+
+    def interpolate(self, what, when, how='interpolate'):
+        df = what
+    
+        df_new = pd.DataFrame.from_dict({'Time' : [when],
+                                    # 'This' : [True]
+                                    })
+        df_new = df_new.set_index('Time')
+    
+        df_combined = pd.concat([df,
+                             df_new],
+                            sort=False)
+        df_combined = df_combined.sort_index()
+        if how == 'interpolate':
```

Comment:
Perhaps it would be nice to add an optional threshold for the age of the document. E.g. if the stuff is 1 week old you get a warning and if its two weeks old you get a ValueError

Reply:
This one is tricky, it would depend on often we take correction data e.g(how often calculate new pmt gains, or how often we measure the elife)

Reply:
throwing this messages is going to be experiment specific